docs/plugins.md

```diff
@@ -0,0 +1,9 @@
+_Authors: grenfeldt@google.com_
+
+# Plugins
+
+The framework will support plugins that run before and after the request handlers. They provide a flexible way to address security issues by setting non-overridable security headers or interrupting and responding to incoming requests before they reach the handler.
```

Comment:
It is okay to write documentation in a somewhat wishful way. i.e. do: "Framework supports plugins..." instead of "Framework *will* support plugins...". Unless this is a design doc, but in this case, this is a user-facing piece of documentation.

Also, make sure you introduce the details in the right places. Not all plugins will run before and after the request handlers, and this is not the level of abstraction that's appropriate. Interceptors (parts of plugins) will do that, so omit it for now.